atomic_reactor/dirs.py

```diff
@@ -93,6 +93,7 @@ def _get_env_from_inspection(data: ImageInspectionData) -> Optional[Dict[str, st
             return envs
         if isinstance(envs, list):
             return dict(item.split("=", 1) for item in envs)
+        assert False, f"Unexpected envs: {envs!r}"
```

Comment:
Maybe this should be a proper error? I would generally expect asserts to be unreachable code

Reply:
True (and they're removed altogether with the `-O`and `-OO` options, and if $PYTHONOPTIMIZE is set... not that we use any of that? But it's a kind of booby trap which could bite us in the future)

My understanding of best practices for Python assertions is "NEVER use them in production code - use Exceptions instead"

Reply:
I think they're OK for asserting something that cannot actually happen at runtime, especially when needed for `mypy` analysis

E.g.
```python
if some_value is None:
    errors.append("some error")
...
if errors:
    raise RuntimeError(...)

assert some_value is not None
```
But here, that is not the case, `ImageInspectionData` is just an alias for `Dict[str, Any]`, so envs can technically be anything

Reply:
I still see this more like development error than runtime error.

IMO envs can be only list or dicts of strings

Reply:
I replaced it with exception to be 100% sure, PTAL